client/src/components/content/RobotContainer/RobotContainer.jsx

```diff
@@ -41,6 +51,34 @@ const RobotContainer = (props) => {
       });
   };
 
+  /**
+   * @description Deletes the robot
+   */
+  const deleteRobot = () => {
+    setConfirmLoading(true);
+    deleteRobotFromDB(robotId).then(() => {
+      setTimeout(() => {
+        setPopConfirmVisible(false);
+        setConfirmLoading(false);
+        removeRobotFromOverview(robotId);
+      }, 1000);
+    });
+  };
+
+  /**
+   * @description Displays the confirmation dialog when user wants to delete a robot
+   */
+  const showPopConfirm = () => {
+    setPopConfirmVisible(true);
+  };
+
+  /**
+   * @description Removes the confirmation dialog when user cancels the deletion process of a robot
+   */
+  const handlePopConfirmCancel = () => {
+    setPopConfirmVisible(false);
+  };
+
```

Comment:
I don't necessarily see the benefit of moving these one-liners out into new functions. From my point of view, we could also just call them directly. If you decide not to do this, please remove the curly braces and return the function directly.

Reply:
Thanks for fixing!